libmattermost.c

```diff
@@ -2288,8 +2275,7 @@ mm_process_msg(MattermostAccount *ma, JsonNode *element_node)
 			}
 		}			
 	} else if (purple_strequal(event, "preferences_changed") && purple_strequal(mm_data_or_broadcast_string("user_id"), ma->self_user_id)) {
-		//hack:  unnamed array of objects that json-glib parser sees as ... string 
-		GList *users = json_array_get_elements(json_named_array_from_string("dont-want-name", json_node_get_string(json_object_get_member(data, "preferences"))));
+		GList *users = json_array_get_elements(json_array_from_string(json_node_get_string(json_object_get_member(data, "preferences"))));

```

Comment:
`json_node_get_string(json_object_get_member(` can be replaced by a single call to `json_object_get_string_member(`